lib/modules/manager/gitlabci/extract.ts

```diff
@@ -76,15 +97,86 @@ export function extractFromJob(
   return deps;
 }
 
+function getIncludeComponentsFromInclude(
+  includeValue: GitlabInclude[] | GitlabInclude,
+): GitlabIncludeComponent[] {
+  const includes = is.array(includeValue) ? includeValue : [includeValue];
+
+  // Filter out includes that dont have a file & project.
+  return includes.filter(isGitlabIncludeComponent);
+}
+function getAllIncludeComponents(
+  data: GitlabPipeline,
+): GitlabIncludeComponent[] {
+  const childrenData = Object.values(filterIncludeFromGitlabPipeline(data))
+    .filter(isNonEmptyObject)
+    .map(getAllIncludeComponents)
+    .flat();
+
+  // Process include key.
+  if (data.include) {
+    childrenData.push(...getIncludeComponentsFromInclude(data.include));
+  }
+  return childrenData;
+}
+function extractDepFromIncludeComponent(
+  includeComponent: GitlabIncludeComponent,
+  endpoint: string | undefined,
+): PackageDependency | null {
+  const componentReference = componentReferenceRegex.exec(
+    includeComponent.component,
+  )?.groups;
+  if (!componentReference) {
+    logger.debug(
+      { componentReference: includeComponent.component },
+      'Ignoring malformed component reference',
+    );
+    return null;
+  }
+  const projectPathParts = componentReference.projectPath.split('/');
+  if (projectPathParts.length < 2) {
+    logger.debug(
+      { componentReference: includeComponent.component },
+      'Ignoring component reference with incomplete project path',
+    );
+    return null;
+  }
+
+  const dep: PackageDependency = {
+    datasource: GitlabTagsDatasource.id,
+    depName: componentReference.projectPath,
+    depType: 'repository',
+    currentValue: componentReference.specificVersion,
+  };
+  if (dep.currentValue === componentReferenceLatestVersion) {
+    logger.debug(
+      { componentVersion: dep.currentValue },
+      'Ignoring component version',
+    );
+    dep.skipReason = 'unsupported-version';
+  }
+  const endpointUrl = parseUrl(endpoint);
+  if (endpointUrl && endpointUrl.hostname !== componentReference.fqdn) {
+    const registryUrl = `https://${componentReference.fqdn}`;
```

Comment:
This will obviously not work if GitLab is served on another port or via HTTP (let's hope nobody does this 😄), but GitLab will run in the same issue when implementing support for this...